src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceResourceIntTest.kt

```diff
@@ -190,13 +196,15 @@ class AttendanceResourceIntTest : IntegrationTestBase() {
       fun `should return bad request if allocation not found`() {
         // Create a new course activity without an allocation
         testData(repository) {
-          offenderBooking = offender(nomsId = "A1234XX") {
-            booking(agencyLocationId = "LEI")
-          }.latestBooking()
           programService {
             courseActivity = courseActivity()
           }
         }
+        testDataFactory.build {
+          offenderBooking = offender(nomsId = "A1234XX") {
+            booking(agencyLocationId = "LEI")
+          }.latestBooking()
+        }
```

Comment:
Next task is to move course activity and its children to `TestDataFactory` at which point we can get rid of `testData` altogether and then think about switching other test suites over to the TestData DSL.